reframe/core/modules.py

```diff
@@ -1002,41 +997,38 @@ def available_modules(self, substr):
         return ret
 
     def _module_full_name(self, name):
-        return self.execute('find', '--format', '{name}-{version}-{hash}',
+        return self.execute('find', '--format', self._name_format,
                             name)
 
     def loaded_modules(self):
-        try:
-            hashes = '/'.join(
-                [m for m in os.environ['SPACK_LOADED_HASHES'].split(':') if m])
-            if hashes:
-                output = self.execute('find', '--loaded', '--format',
-                                      '{name}-{version}-{hash}', f'/{hashes}')
-                return [Module(m) for m in output.split('\n') if m]
-            else:
-                return []
-        except KeyError:
+        hashes = '/'.join(
+            [m for m in os.environ['SPACK_LOADED_HASHES'].split(':') if m])
+        if hashes:
+            output = self.execute('find', '--loaded', '--format',
+                                  self._name_format, f'/{hashes}')
+            return [Module(m) for m in output.split('\n') if m]
+        else:
             return []
 
     def conflicted_modules(self, module):
-        pass
+        return []
 
     def is_module_loaded(self, module):
         module = self._module_full_name(module.name)
         module = Module(module)
         return module in self.loaded_modules()
 
     def load_module(self, module):
-        pass
+        self.execute('load', '--sh', str(module))
```

Comment:
This has no effect. This will produce the shell commands to load the module. This is not what we want by a module system. We would want the Python bindings, but Spack does not offer any, so we should simply return the same as the `nomod` backend here.